endpoint.go

```diff
@@ -222,6 +228,15 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+func (ep *endpoint) needDNSServer() bool {
+	if ep.getNetwork().Type() == "null" || ep.getNetwork().Type() == "host" {
+		return false
+	}
```

Comment:
not really. we try hard to keep libnetwork core driver independent. (even if it is a built-in one).